# 공유 Qdrant에 어떤 collections들 있는지 확인용

## 리스트 
(밑에 셀에 다 삭제하는 코드 있음)

In [18]:
import os
from dotenv import load_dotenv
from qdrant_client import QdrantClient

# Load Qdrant credentials from the project .env file
load_dotenv("../.env")

client = QdrantClient(
    url=os.environ["QDRANT_URL"],
    api_key=os.environ["QDRANT_API_KEY"],
)

collections_resp = client.get_collections()

total_cnt = 0
for coll in collections_resp.collections:
    total_cnt += 1
    info = client.get_collection(coll.name)  # CollectionInfo
    vectors = info.vectors_count or {}
    total_vectors = sum(v.count for v in vectors.values()) if vectors else "n/a"

    print(
        f"- {coll.name} | points: {info.points_count} | vectors: {total_vectors}"
    )

print(f"Total # of collections: {total_cnt}")

- trade_collection | points: 1048 | vectors: n/a
Total # of collections: 1


trade_collection: 

certification 454 points 
-> claim 948 points  (추가 확인)
-> cisg 1048 points (추가 확인)



## 삭제용

In [4]:
def delete_all_collections():
    """Delete all collections in Qdrant."""
    collections_resp = client.get_collections()
    
    deleted_count = 0
    for coll in collections_resp.collections:
        try:
            client.delete_collection(coll.name)
            print(f"✓ Deleted collection: {coll.name}")
            deleted_count += 1
        except Exception as e:
            print(f"✗ Failed to delete {coll.name}: {e}")
    
    print(f"\nTotal collections deleted: {deleted_count}")


def delete_target_collection(collection_name: str):
    """
    Delete a specific collection in Qdrant.
    
    Args:
        collection_name: Name of the collection to delete
    """
    try:
        client.delete_collection(collection_name)
        print(f"✓ Successfully deleted collection: {collection_name}")
    except Exception as e:
        print(f"✗ Failed to delete collection '{collection_name}': {e}")

### 원하는 거 하나만 지우기

In [16]:
delete_target_collection("cisg_document_collection")

✓ Successfully deleted collection: cisg_document_collection


### 전부 지우기

In [11]:
delete_all_collections()  # ⚠️ WARNING: This will delete ALL collections!

✓ Deleted collection: trade_collection

Total collections deleted: 1
